scripts/covid19_india/cases_count_districts_data/preprocess.py

```diff
@@ -0,0 +1,75 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#         https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import pandas as pd
+import difflib
+
+"""
+Retrieving and preprocessing district-wise daily case counts for India.
+Raw data retrieved using API by covid19india.org (https://www.covid19india.org/)
+"""
+
+
+output_columns = [
+    'Date', 'District', 'districtCode',
+    'Count_MedicalTest_ConditionCOVID_19_Positive',
+    'Count_MedicalConditionIncident_COVID_19_PatientRecovered',
+    'Count_MedicalConditionIncident_COVID_19_PatientDeceased'
+]
+
+
+def _get_district_code(district):
+    dist_code = pd.read_csv('../common/data/districts-local-directory.csv')
+    if district not in ['Unknown']:
+        close_match = difflib.get_close_matches(str(district).upper(), dist_code['DistrictName(InEnglish)'],
+                            n=1, cutoff=0.8)
+        if close_match:
+            return dist_code[dist_code['DistrictName(InEnglish)'] == close_match[0]]['DistrictCode'].values[0]
+        else:
+            return None
+
+
+def create_formatted_csv_file(csv_file_path, df):
+    # creating dictionary with district vs code mappings
+    district_codes = []
+    
+    for district in df['District'].unique():
+        district_codes.append(_get_district_code(district))
+    dist_codes_dict = dict(zip(df['District'].unique().tolist(), district_codes))
+    
+    #get the district codes and drop rows that do not have district codes
+    df['districtCode'] = df.apply(lambda row: dist_codes_dict[row['District']],
+                                                axis=1)
```

Comment:
how many do we drop?

Reply:
61 districts from the covid19india.org data (643 -> 582). I added a comment in the code mentioning the same.